data-prepper-plugins/kafka-source/src/main/java/org/opensearch/dataprepper/plugins/kafka/source/configuration/ConsumerConfigs.java

```diff
@@ -0,0 +1,244 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.dataprepper.plugins.kafka.source.configuration;
+
+import java.time.Duration;
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+/**
+ * * A helper class that helps to read user configuration values from
+ * pipelines.yaml
+ */
+public class ConsumerConfigs {
+
+  private static final String AUTO_COMMIT = "false";
+  private static final Duration AUTOCOMMIT_INTERVAL = Duration.ofSeconds(1);
+  private static final Duration SESSION_TIMEOUT = Duration.ofSeconds(45);
+  private static final int MAX_RETRY_ATTEMPT = 5;
+  private static final String AUTO_OFFSET_RESET = "earliest";
+  private static final Duration THREAD_WAITING_TIME = Duration.ofSeconds(1);
+  private static final Duration MAX_RECORD_FETCH_TIME = Duration.ofSeconds(4);
+  private static final Duration BUFFER_DEFAULT_TIMEOUT = Duration.ofSeconds(5);
+  private static final Duration MAX_RETRY_DELAY = Duration.ofSeconds(1);
+  private static final Long FETCH_MAX_BYTES = 52428800L;
+  private static final Long FETCH_MAX_WAIT = 500L;
+  private static final Long FETCH_MIN_BYTES = 1L;
+  private static final Duration RETRY_BACKOFF = Duration.ofSeconds(100);
+  private static final Duration MAX_POLL_INTERVAL = Duration.ofSeconds(300000);
+  private static final Long CONSUMER_MAX_POLL_RECORDS = 500L;
+  private static final Integer NUM_OF_WORKERS = 3;
+  private static final Duration HEART_BEAT_INTERVAL_DURATION = Duration.ofSeconds(3);
+
+  @JsonProperty("group_id")
```

Comment:
This should be marked as `@NonNull`. (Also, see my comments above about moving up.